src/plugins/event-plugins/JsErrorPlugin.ts

```diff
@@ -62,10 +65,15 @@ export class JsErrorPlugin implements Plugin {
     }
 
     private eventHandler = (errorEvent: ErrorEvent) => {
-        this.recordEvent(
-            JS_ERROR_EVENT_TYPE,
-            errorEventToJsErrorEvent(errorEvent, this.config.stackTraceLength)
-        );
+        if (!this.config.shouldIgnoreError(errorEvent)) {
```

Comment:
Also doing an early exit instead of wrapping in an if statement may be better here

Reply:
are you suggesting something in the lines of this?:
```
private promiseRejectEventHandler = (event: PromiseRejectionEvent) => {
        if (this.config.filter(event)) {
            return;
        }
        this.recordJsErrorEvent({
            type: event.type,
            error: event.reason
        } as ErrorEvent);
    };
```